fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3379,7 +3379,7 @@ public List<LoanTransaction> retrieveListOfTransactionsByType(LoanTransactionTyp
     private boolean doPostLoanTransactionChecks(final LocalDate transactionDate,
             final LoanLifecycleStateMachine loanLifecycleStateMachine) {
         boolean statusChanged = false;
-        if (isOverPaid()) {
+        if (getTotalOverpaid() != null && getTotalOverpaid().compareTo(BigDecimal.ZERO) > 0) {
```

Comment:
I think this condition could be extracted at least to a variable. It makes the code much easier to read.

Reply:
sure